src/containers/App/App.js

```diff
@@ -21,40 +22,91 @@ import {
 } from 'react-router-dom';
 
 import {
+  Extension,
+  Extensions,
   Home,
   PipelineRun,
   PipelineRuns,
   Pipelines,
   Tasks,
   TaskRuns
 } from '..';
+import { fetchExtensions } from '../../actions/extensions';
+import { getExtensions } from '../../reducers';
 
 import '../../components/App/App.scss';
 
-const App = () => (
-  <Router>
-    <Switch>
-      <Route path="/" exact component={Home} />
-      <Redirect
-        from="/pipelines/:pipelineName"
-        exact
-        to="/pipelines/:pipelineName/runs"
-      />
-      <Redirect from="/tasks/:taskName" exact to="/tasks/:taskName/runs" />
-      <Route path="/pipelines" exact component={Pipelines} />
-      <Route path="/tasks" exact component={Tasks} />
-      <Route
-        path="/pipelines/:pipelineName/runs"
-        exact
-        component={PipelineRuns}
-      />
-      <Route path="/tasks/:taskName/runs" exact component={TaskRuns} />
-      <Route
-        path="/pipelines/:pipelineName/runs/:pipelineRunName"
-        component={PipelineRun}
-      />
-    </Switch>
-  </Router>
-);
+export class App extends Component {
+  componentDidMount() {
+    this.props.fetchExtensions();
+  }
+
+  render() {
+    const { extensions } = this.props;
+
+    return (
+      <Router>
+        <Switch>
+          <Route path="/" exact component={Home} />
+          <Redirect
+            from="/pipelines/:pipelineName"
+            exact
+            to="/pipelines/:pipelineName/runs"
+          />
+          <Redirect from="/tasks/:taskName" exact to="/tasks/:taskName/runs" />
+          <Route path="/pipelines" exact component={Pipelines} />
+          <Route path="/tasks" exact component={Tasks} />
+          <Route
+            path="/pipelines/:pipelineName/runs"
+            exact
+            component={PipelineRuns}
+          />
+          <Route path="/tasks/:taskName/runs" exact component={TaskRuns} />
+          <Route
+            path="/pipelines/:pipelineName/runs/:pipelineRunName"
+            component={PipelineRun}
+          />
+          <Route path="/extensions" exact component={Extensions} />
+          {extensions.map(({ displayName, name, source }) => {
```

Comment:
move the conditional for display name to a filter ?

Reply:
are you thinking something like this?

```js
extensions
  .filter(({ displayName }) => !!displayName)
  .map(({ displayName, name, source }) => (
    <Route
      key={name}
      path={`/extensions/${name}`}
      render={({ match }) => (
        <Extension
          displayName={displayName}
          match={match}
          source={source}
        />
      )}
    />
  ));
```

I went with including it in the `map` originally as it's just a single pass over the extensions (taking advantage of the fact that React skips `nulls` when rendering), but since it should be a relatively small list I'm ok with either.